src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndex.cs

```diff
@@ -34,7 +34,12 @@ public void Add(IFunctionDefinition function, FunctionDescriptor descriptor, Met
             _functionsByMethod.Add(method, function);
 
             // For compat, accept either the short name ("Class.Name") or log name (just "Name")
-            _functionsByName.Add(descriptor.LogName, function);
+            if (!_functionsByName.ContainsKey(descriptor.LogName))
```

Comment:
This fix retains the previous logic of first one wins that existed before cache was added. See previous logic here: [here](https://github.com/Azure/azure-webjobs-sdk/commit/76d44ca915b817ed3f3cc46e355aa6cb0f741b4b#diff-1063e70146286d93a5d60e9e0c2cf6043c3fd84e07199fec91bc585f8b087eeaL54).